src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+                bzero(data, INSTANCE_DATA_LENGTH + 1);
+                read(sockfd, data, INSTANCE_DATA_LENGTH);
+                response = (data);
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string workerList = (data);
-            workerList = Utils::trim_copy(workerList, " \f\n\r\t\v");
-            instance_logger.log("Received Worker List " + workerList, "info");
-
-            std::vector<string> workerSockets;
-            stringstream wl(workerList);
-            string intermediate;
-            while (getline(wl, intermediate, ',')) {
-                workerSockets.push_back(intermediate);
+                if (response.compare(JasmineGraphInstanceProtocol::FILE_RECV_CHK) == 0) {
+                    write(sockfd, JasmineGraphInstanceProtocol::FILE_RECV_WAIT.c_str(),
+                          JasmineGraphInstanceProtocol::FILE_RECV_WAIT.size());
+                }
             }
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
-
             bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphVertexCount = (data);
-            graphVertexCount = Utils::trim_copy(graphVertexCount, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID:" + graphID + " Vertex Count: " + graphVertexCount, "info");
+            read(sockfd, data, INSTANCE_DATA_LENGTH);
+            response = (data);
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+            if (response.compare(JasmineGraphInstanceProtocol::FILE_RECV_CHK) == 0) {
+                instance_logger.log("Received : " + JasmineGraphInstanceProtocol::FILE_RECV_CHK, "info");
+                write(sockfd, JasmineGraphInstanceProtocol::FILE_ACK.c_str(),
+                      JasmineGraphInstanceProtocol::FILE_ACK.size());
+                instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::FILE_ACK, "info");
+            }
+
+            Utils::unzipDirectory(fullFilePath);
+            size_t lastindex = fileName.find_last_of(".");
+            string pre_rawname = fileName.substr(0, lastindex);
+            size_t next_lastindex = pre_rawname.find_last_of(".");
+            string rawname = fileName.substr(0, next_lastindex);
+            fullFilePath =
+                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.trainedmodelfolder") + "/" + rawname;
 
+            while (!Utils::fileExists(fullFilePath)) {
+                bzero(data, INSTANCE_DATA_LENGTH + 1);
+                read(sockfd, data, INSTANCE_DATA_LENGTH);
+                string response = (data);
```

Comment:
Do we need parenthesis here?

Reply:
This is not present after wrapping read in next PR

Reply:
Ok